### ISSUE TITLE
Fails when removing multiple elements from array

### ISSUE THREAD

**Issue by aaproducto** [swaggest/json-diff#6]
When you remove more than one element from an array, it breaks. 

A test case: 

source
```
[{"name":"a"},{"name":"b"},{"name":"c"}]
```
target:
```
[{"name":"a"}]
```
Error: 
```
Uncaught exception 'Swaggest\JsonDiff\Exception' with message 'Key not found: 2' in /var/www/alquilerargentina/vendor/swaggest/json-diff/src/JsonPointer.php:185
```

note: pretty similar to this one https://github.com/java-json-tools/json-patch/issues/11

**Comment by vearutop** [comment]
There is indeed a misbehaving, when I apply 
```
[{"op":"remove","path":"/0"},{"op":"remove","path":"/2"}]
``` 
I get `{"1":{"name":"b"}}` instead of `[{"name":"b"}]`. I'll fix that.

But back to your error, I was unable to reproduce it. 

`swaggest/json-diff` produces 
```
[
    {
        "value": "a",
        "op": "test",
        "path": "/0/name"
    },
    {
        "value": "b",
        "op": "replace",
        "path": "/0/name"
    },
    {
        "op": "remove",
        "path": "/1"
    },
    {
        "op": "remove",
        "path": "/2"
    }
]
```
from your two jsons, which maybe isn't very short, but applies without error.

Could you share exact code that produced exception?

**Comment by aaproducto** [comment]
This is the code that produced the error:
```php
$originalJson = '[{"name":"a"},{"name":"b"},{"name":"c"}]';
$newJson = '[{"name":"b"}]';
$diff = new JsonDiff(json_decode($originalJson), json_decode($newJson));

//we need to serialize the patch to send it to AWS SQS
$patchJson = json_decode( json_encode($diff->getPatch()->jsonSerialize()), true );

$original = json_decode($originalJson);
$patch = JsonPatch::import($patchJson);
$patch->apply($original);
```
This is a simple test to reproduce an error that we having in our project.
By the way the version i am using is the v2.0.1

**Comment by vearutop** [comment]
Should be fixed with `v2.0.2`
